src/settings/SettingsView.ts

```diff
@@ -448,7 +448,14 @@ export class SettingsView extends BaseTopLevelView implements TopLevelView<Setti
 	oncreate(vnode: Vnode<SettingsViewAttrs>) {
 		locator.eventController.addEntityListener(this.entityListener)
 
-		this.populateAdminFolders()
+		const currentRoute = m.route.get()
+		this.populateAdminFolders().then(() => {
+			// We have to wait for the folders to be initialized before setting the URL,
+			// otherwise we won't find the requested folder and will just pick the default folder
+			if (currentRoute !== this._userFolders[0].url) {
```

Comment:
okay, sorry but I still don't get why we only do it if the URL is *not* the first one. I thought we only want to switch if it *is* the first (default) one? Otherwise we might be elsewhere already?

Reply:
When users navigate to settings from an internal link coming from an email for example, the onNewUrl is triggered moving they to the default route because not all the folders are available yet, **but** the currentRoute remains the internal link that they want to access, to avoid an unnecessary navigation, like moving again to the default one, the check is realized

Reply:
I think I finally get it! I was confused because I forgot that `currentRoute` already has the folder param. I don't think it's harmful to change URL to the same one that we already have, it's probably a noop?

So what I think we actually should do it is

```ts
const stillAtDefaultUrl = m.route.get() === this._userFolders[0].url
if (stillAtDefaultUrl) {
  // the user did not navigate away yet, it is safe to switch to the folder from the parameter
  // ...
}
```

another thing that I've noticed is that we are doing `m.route.param("folder")` but *after* the async loading. Shouldn't we save the folder parameter to a variable before loading (before it's possibly switched)?